docs/alternatives.rst

```diff
@@ -0,0 +1,132 @@
+which means that they must be deployed there.
+
+Kopf, on the other hand, attempts to keep the things explicit
+(as per the `Zen of Python`_: *explicit is better than implicit*),
+keeping the whole operator's logic in one place, in one syntax (Python).
+
+.. admonition:: And, by the way...
+
+    Not only it is about "*explicit is better than implicit*",
+    but also "*simple is better than complex*",
+    "*flat is better than nested*", and "*readability counts*",
+    which makes Kopf a *pythonic* framework in the first place,
+    not just *written with Python*.
+
+Kopf also makes the effort to keep the operator development human-friendly,
+which means at least the ease of debugging (e.g. with the breakpoints,
+running in a local IDE, not in the cloud), the logs readability,
+and other little pleasant things.
+
+And also Kopf allows to write *any* arbitrary domain logic of the resources,
+especially if it spans over the long time periods (hours, days if needed),
+and is not limited to the timeout restrictions of the HTTP APIs with their
+expectation of nearly-immediate outcome (i.e. in seconds or milliseconds).
+
+Metacontroller, however, is more mature, 1.5 years older than Kopf,
+and is backed by Google, which originally developed Kubernetes itself.
+
+Unlike Kopf, Metacontroller supports the domain logic in any languages
+due to its language-agnostic nature of HTTP APIs.
+
+.. _Metacontroller: https://metacontroller.app/
+.. _Zen of Python: https://www.python.org/dev/peps/pep-0020/
+
+
+Side8's k8s-operator
+====================
+
+Side8's k8s-operator_ is another direct equivalent.
+It was actually the initial inspiration for writing Kopf.
+
+Side8's k8s-operator is written with Python3, and allows to write
+the domain logic in the apply/delete scripts in any language.
+The scripts run locally on the same machine where the controller is running
+(usually the same pod, or a developer's computer).
+
+However, the interaction with the script relies on stdout output
+and the environment variables as the input,
+which is only good if the scripts are written in shell/bash.
+Writing the complicated domain logic in bash can be troublesome.
+
+The scripts in other languages, such as Python, are supported, but require
+the inner infrastructure logic to parse the input and to render the output
+and to perform the logging properly:
+e.g., so that no single byte of garbage output is ever printed to stdout,
+or so that the resulting status is merged with the initial status, etc --
+which kills the idea of pure domain logic and no infrastructure logic
+in the operator codebase.
+
+.. _k8s-operator: https://github.com/side8/k8s-operator
+
+
+CoreOS Operator SDK & Framework
+===============================
+
+`CoreOS Operator SDK`_ is not actually an operator framework.
+It is an SDK, i.e. a Software Development Kit,
+which generates the skeleton code for the operators-to-be,
+and then the developers should enrich it with the domain logic code as needed.
```

Comment:
"and users should enrich" ... I guess here "developers" would mean the developers of the SDK :)